### ISSUE TITLE
Make Resting in the Inn Information Available and Obvious Outside of the Tavern (was: Stuck in yesterday -- Resting in Inn -- Dailies did not reset)

### ISSUE THREAD

**Issue by IcanBEsoMUCHbetter** [HabitRPG/habitica#4742]
_edit by admin:_
### To everyone who experienced this problem, please see the first comment below (the one that is all in bold by Alys).

_end of edit by admin_

---

Hi!

User ID : 8f9ff296-d9c2-401c-aedb-1fe20c74ccc3
OS : Windows XP
Browser : Chrome

I keep on Refreshing the page, nothing happens. The damage I did to the boss quest from yesterday won't appear, the dailies for today won't appear, the damage I took for not doing some of yesterday's dailies won't appear either. The data display shows I already received 4 drops today while doing nothing but when I tried to use one of them, a message box appeared saying that there was an error processing at the moment and to contact GitHub. The message box then vanished.

The XP graph won't show me data for 24/02/2015. It stops at 23/02/2015.

I also woke up to see that I was sleeping at the Tavern, but never once went there either.

Everything is so confusing and I can't find an issue similar to this one on here.

Thanks so much for the help :)

**Comment by Alys** [comment]
**We had major server problems for several hours over the past day, so we caused all active players to rest in the inn, to minimise the number of people who died due to not being able to complete their Dailies during the outage. We are sorry that it also will have prevented completed Dailies being unticked, but we felt it was the best option overall. I understand your frustration though. :(**

**You can check out of the inn from the left-hand side of the Tavern page on the website, or from the Settings screen in the app.**

**If you'd like to use your Dailies today without losing stats, make a note of your current Experience (XP) and Gold (GP), then untick all your Dailies. Then go to [Settings -> Site](https://habitrpg.com/#/options/settings/settings) and click on the Fix Character Values button. You will be shown a form that you can use to set your XP and GP back to what they were before you unticked the Dailies.**

**Comment by IcanBEsoMUCHbetter** [comment]
Wow, thank you so much for the very quick reply. I didn't realize you guys
were in trouble yesterday, I was too busy being productive (thanks to
HabitRPG....damn, that thingy is GOOD). I'll go do as you suggested.

Have a nice one :D

2015-02-24 7:43 GMT-05:00 Alice Harris notifications@github.com:

**Comment by crookedneighbor** [comment]
So, I have an idea for an Inn-provement. What if when cron runs, it checks if the user is in the inn, and if so, opens up a modal asking if the user would like to check out? 

For users using the inn because they won't be able to log in for a few days (sickness, vacation, etc), it won't affect them until the next time they log in, and would be a nice shortcut to get going again.

For users that are in the inn, but using the site to still check off todos and mark habits on a daily basis, they'd only be bugged about checking out of the inn once a day.

Thoughts @lemoness @Alys @sugarfiend?

**Comment by Alys** [comment]
@crookedneighbor I think it's an EXCELLENT idea but I am rejecting it completely because of the pun. :)

What will happen though if cron runs from an action that a user does not perform themselves? For example, an automatic Beeminder goal update will trigger cron, as would any other third-party access. Should cron actually set a showCronModal flag and then the flag triggers the modal to appear, in a Bailey-like fashion?

I personally would like to a see it have a "don't ask me again" option (setting a preference, which would be cleared when they do check out of the inn, so for your next inn visit you'd have to select that option again if you still wanted it). There'll be plenty of cases where people are in the inn while still using HabitRPG, and I think some would find it annoying to have modals each day. I've seen complaints about the frequency of Baileys, and that's only a couple of times a week.

FYI to anyone querying this: cron DOES run once a day when you are in the Inn, but one of its early tasks is to see if you are in the Inn and then exit if you are. So this is certainly possible!

ping @anitaycheng (sugarfiend)

**Comment by Alys** [comment]
ping @anitaycheng (sugarfiend)

**Comment by crookedneighbor** [comment]
Yes, I was thinking of a showCronModalFlag that would behave like Bailey. I agree, an opt out option would be good. 

Sorry @anitaycheng, I didn't know your Github Username.

**Comment by lemoness** [comment]
I really like this, but an opt-out button like "Don't show me these
messages again" is crucial. But yes, we get questions about this a LOT.

How would this interact with the future plan to record yesterday's
activity? Any intersection at all, either user-side or code-side?

**Comment by deilann** [comment]
I think it's also important to refresh a player's tasks after checking out
of the inn.

The reason we didn't do it before was because we were worried about people
exploiting it... but we're a lot less worries about exploits these days and
it would make the inn a lot more of a useful feature.

**Comment by crookedneighbor** [comment]
People do seem to expect for the dailies to reset upon exiting the inn, so I'd say that is a good idea.

**Comment by lemoness** [comment]
It would only refresh on cron, right? If so, I'm in favor.

**Comment by AnitaYCheng** [comment]
@crookedneighbor Yeah I signed up for GitHub for a hackathon, hence my serious name. :smile: 

I don't think we need a modal for this. Modals should be reserved for level-ups and our future yesteradailies features only, in my opinion, because it directly affects gameplay. Level ups in particular should be celebrated as a big win. And we can wish them good morning/afternoon/evening in the yesterdaily. :smile: 

My idea is a popover on the avatar, reminding them that they're in the inn and give them a link for them to check out. 

We can have a little bouncing/animate-once notification for Bailey too. The frequent modals for Bailey also annoy me sometimes, haha. (Sorry Bailey!) 

In both instances, it calls attention to the item without disrupting flow.

**Comment by deilann** [comment]
A modal every time you leveled up would annoy me.

**Comment by AnitaYCheng** [comment]
@deilann We should at least have a level-up modal when big features are introduced, for sure!

When you get up to the higher levels where nothing much changes, a centered alert would suffice, and it's something we should test. It's a bigger deal than a random drop, is the point.

**Comment by deilann** [comment]
When big features are introduced, we have Justin, which is an interactive
modal, for all intents and purposes.

A centered alert would disrupt my work flow and be rather annoying. I don't
need something I need to click to continue.

**Comment by AnitaYCheng** [comment]
@deilann I'm talking about an alert that goes away by itself, like what drops are right now.

My concern is, when the current "Level up" alert appears, it's so small and inconspicuous that when Justin shows up, the player doesn't associate the new features as strongly with levelling up. Especially when they're a new player. Yeah, when you're up on the higher levels, it's not as big of a deal, but before they get used to it, we need to give them all the encouragement they can get.

I suspect this also depends on how the player uses HabitRPG and how fast they level up. If it's multiple times per day, or once a day, yeah I can see how that would get annoying. (Personally, I do it once or twice a week, which isn't that bad.) Is there any way to pull stats on this @lefnire @crookedneighbor?

**Comment by deilann** [comment]
That's a growl, not a modal. A modal window, by definition, does not go
away by itself. See: http://en.wikipedia.org/wiki/Modal_window

**Comment by AnitaYCheng** [comment]
Bootstrap seems to call our drops alerts. -shrug- http://getbootstrap.com/components/#alerts

I still think early players should get a level-up modal, to be more direct about showing them around. At least until they start getting quest drops. Once they hit that, then the only thing that changes is amount of MP and XP they can fill, and special types of quest drops. That can easily fit into a growl-type alert.

**Comment by Alys** [comment]
@AnitaYCheng The database doesn't record historical information such as when level-ups occurred. We could pull everybody's level and length of time that they had been using HabitRPG, and average the level ups, but a lot of factors would make that meaningless (Rebirths, extended breaks from using HabitRPG, etc). I'm not sure that Google Analytics or other stats could help either because of the way leveling is handled.

I'd really prefer to not have a growl in the centre of the screen. The ones that cover the rewards are annoying enough at times (but I am not suggesting we remove them); I definitely wouldn't want a growl covering tasks as I was using them, even for a short time. Leveling up tends to be obvious from other factors (the Health bar for instance, when you're low on Health, and always the XP bar). 

Random drops are often a bigger deal for me than level ups. Drops are exciting.

The need for a Resting in the Inn notification is that there are a LOT of players who aren't realising that they're Resting. We're getting very frequent comments in the Tavern asking why their Dailies aren't resetting. This is a major interruption to their usage of HabitRPG - one extra modal is nothing by comparison. Even before we did a forced Rests, there were occasional quests from users who didn't know they were Resting or didn't understand what it means.

A popover on the avatar sounds interesting, but I'd want it to be extremely obvious. Many players are ignoring the Baileys, so something of that nature isn't enough to notify them about the enforce Rest.

**Comment by Alys** [comment]
I'm not against more obvious introductions to new features that you get on level-up, but it's a very separate, and less urgent issue.

**Comment by AnitaYCheng** [comment]
@Alys Thanks for the info! Yeah, this is why I need lots more respondents on the usage survey, so we can see exactly what people find motivating about playing Habit. :wink: Personally, I check things off so I don't die, so leveling up to fill my health back up is a big deal for me, but I'm not everyone, and neither are the (relatively) few of us here. 

My idea for the avatar popover is more direct than a modal, and I suspect that people would start ignoring all modals because they are already ignoring Bailey. I believe many people ignore Bailey because it's a wall of what they perceive to be largely irrelevant text, and such people likely just dismiss it every time without even looking at it.

If it's a colored popover directly on the avatar that does some kind of flashy animation that gets their attention, we could also educate them that the ZZZ's over their heads means they're in the inn. A modal wouldn't do that very well.

**Comment by Alys** [comment]
Fair point about the modal. How do you see the popover working? When does it appear / what triggers it? How long does it remain for? What options does it give the users for interacting with it?

**Comment by deilann** [comment]
Another good way to figure out what people find motivating about Habit is
to frequent the Tavern, the Newbies guild, the Chronic Illness guild, the
ADHDers guild... Reading what the users talk about and what help they need
will give a different view than a survey. There's a lot of good info out
there that you're going to miss with a self-selected survey.

**Comment by AnitaYCheng** [comment]
@deilann In [the survey](https://www.surveymonkey.com/s/62RXRF3) (which you should take if you haven't already :wink: ), I actually have each respondent rank 10-ish different features, so it's different than just trawling through guilds/social media looking for mentions. I have done stuff like that before, when I didn't have direct access to users, but that method lacks information on context of use, which is extremely important. (A few mentions I've seen on Twitter get very specific about context of use, but it's been less than 5 out of maybe 100-200 mentions...I can't be bothered to count how many stickies I have on my Mural.ly board right now...) We ask that in the survey as well.

For example, maybe they find quests very motivating and notable for mention, but they really like collecting pets too. Whereas another person might find quests motivating and notable for mention but not care for pets at all. It isn't all or nothing, and we need to get a big picture.

@Alys, I imagine the popover would be triggered once they logged into their tasks dashboard and they're in the inn. (Which was the idea for the modal, I believe?) It would persistently stay on partly over the avatar until it's either dismissed or they click the "Check out of the inn" button. To grab attention, it could bounce or flash or do something every 10-20 seconds (should be tested), and if it partly covers the Habits column (just enough to cover the title), I think most people would notice it eventually. How should we test that theory? Since it's connected to a user-generated action, I don't think we could build a wireframe then ask people if they notice anything different, because that's a leading question, haha.

**Comment by Alys** [comment]
So the popover would be basically very much like the Bailey modal except that it would cover only a small part of the screen and would allow the users to interact with the rest of the screen without first making the popover go away?

One thing to keep in mind is that some users have the header scrolled off the top of the screen or permanently hidden, so tying it to the avatar specifically wouldn't be reliable.

**Comment by AnitaYCheng** [comment]
@Alys, I think having the header hidden would be ok, although we might need to change the wording if we can't show them the Zzzs in the avatar. It would just be positioned somewhere near where the avatar would be, so they at least know the avatar should show them what's going on.

Whipped up a mockup of what I think it could look like: 
![checkoutofinnpopover](https://cloud.githubusercontent.com/assets/6760632/6407724/a0635fe4-bdfa-11e4-8bc3-c2796b6a62ff.jpg)

**Comment by deilann** [comment]
Yeah. I took that survey and found that part kind of useless. Psychology
shows that it's really hard to rank that many things together. The longer
the list, the more likely people are to kind of just randomize them a bit.
The initial list order also greatly influences what people will say,
generally. It also doesn't allow me to tell you anything beyond ranking
them in order. It doesn't let me tell you how motivating I find each one.
It also assumes that I'm aware of my preferences, which psychology has also
shown is usually not entirely the case.

http://faculty.haas.berkeley.edu/ned/Choice_Fatigue.pdf
http://link.springer.com/article/10.1007/BF01098791
http://www.chicagobooth.edu/research/workshops/marketing/archive/WorkshopPapers/vohs.pdf

I really do suggest that you interact with the users more. It's up to you,
but it you might not be in touch with how the most active portion of the
userbase interacts with the site, especially if you aren't aware of the
biggest problems we're having at the moment. It'll get you raw information
that's less "self-reported" and more a snapshot of their knee-jerk,
emotional reaction, rather than an analytical sit-down with a survey.

**Comment by Alys** [comment]
@AnitaYCheng Suggested changes for the popover:

Title: **You're Resting in the Inn**  (I think the full term is important to help them recognise it)
First line: **The Zzz's over your avatar's head means you are [Resting in the Inn](http://habitrpg.wikia.com/wiki/Tavern#Resting_in_the_Inn).**
Second line: **Your Dailies won't reset until you check out.** (The same but with the usual capitalisation.)
First button: **I'm still resting but remind me tomorrow**
Second button: **Don't remind me again**
Third button: **Check out of the Inn**

**Comment by AnitaYCheng** [comment]
@deilann, the only way to interact with the players and being able to connect with them on the level that you suggest, is to interview them one by one. That's the only way to get the total picture (instead of limited self-reports and superficial survey answers) and connect all the dots. Which is totally possible, of course, given the right incentive structure for both the interviewer and interviewee. :wink: It would be a deep dive into relatively few users, though - only large corporations have the budget/time/personnel to talk to hundreds of users.

The Newbies guild is where the biggest, most obvious problems show up, and I think we do want to focus making things easier for the newbies right now. And the survey was the fastest/easiest way to get a big picture of what people are using the site for in general. (The ranking list is randomized for each person, so that's one thing..) Once we've put the fires out (so to say - there's a lot of usability issues that are low hanging fruit), I will start another giant Mural.ly board with all the questions in Tavern. :grinning: I'm still recovering from my Twitter one...

**Comment by AnitaYCheng** [comment]
@Alys, how's this? Even with the block buttons, it still doesn't cover the tasks, hah.

![checkoutofinnpopover1](https://cloud.githubusercontent.com/assets/6760632/6407930/0e2f6b68-bdff-11e4-8185-e831a8c48ef0.jpg)

**Comment by Alys** [comment]
@AnitaYCheng Yes, I like that popover. We should leave out the little white arrow anchor thing at the top because that will look odd when the avatar is hidden, and I would suggest tying it to the task button. That will hide the avatar for those who have the avatar showing (and so they'll really notice it!) and it should still hide the top of the Habits list for other users. But I'd definitely want to hear Blade's and Lemoness's thoughts. These so far are just mine. :)

Regarding user opinions: I believe @deilann is talking about something that's pretty much the opposite of interviewing them. An interview or a survey will tell you their thoughts when they are thinking calmly and rationally about HabitRPG, but it won't tell you anything about how they think on a day to day basis, especially not when they're stressed from a bug or from a feature that they don't understand. You can get a feel for that sort of thing by regularly reading the Tavern for weeks - you pick up patterns, notice common feelings. It sounds like you might be reading the Newbies guild regularly, which is a very good idea, but I'd suggest the Tavern as well since there's many comments from established users there, and also the two other guilds deilann mentioned for users with special requirements.

**Comment by AnitaYCheng** [comment]
@Alys and @deilann, I do think it has to be a mix of both. People often react very negatively in the moment out of frustration, but their experience overall can still be good. (I have seen this myself numerous times in other communities.) That doesn't mean we need to be catering to everything mentioned on a code-level, especially when people have different priorities about what they're doing on the site. What we should be most concerned with, regarding building a product, is whether players can easily do the most important things that the vast majority of people want to do (especially important if they're paying subscribers), and if they can find their way to more advanced features if it isn't immediately visible. And again, there's still a lot of things that need to be done in the meantime!

**Comment by Alys** [comment]
@AnitaYCheng I don't think we should consider the needs of paying subscribers to be more important than other users. We've always had the concept of HabitRPG not being "pay to win" and so we need to treat all users equally. I feel that very strongly.

**Comment by deilann** [comment]
regarding the comment on subscribers: This is an open source project that
works as hard as possible to make the experience of a free user as
important as the experience of a paid user. Historically, Habit has not
given preference to what subscribers want, because that is against our
values.

And here's the thing: it does need to be both, but your comments make it
seem like you're not doing both. You don't seem to be aware of what Tavern
culture is like, what users post about, and what makes them excited enough
to share. I'm not saying "don't do surveys;" I'm saying "start reading the
Tavern."

**Comment by lemoness** [comment]
I like that popover a lot! I wonder if there's a way we can also edit the text around Daniel's Rest in the Inn speech bubble to make it more punchy and/or obvious what it will do. Right now it says:

Welcome to the Tavern! Stay a while and meet the locals. If you need to rest (vacation? illness?), I'll set you up at the inn. While checked-in, your Dailies are frozen as-is (checked/unchecked) until the day after check-out. You will not suffer for missing them at the day's end.

REST IN THE INN

Be warned: If you are participating in a boss quest, the boss will still damage you for your party mates' missed Dailies!

.--

Maybe something roughly like:

Welcome to the Tavern! Here you can:
- Chat with the community!
- Rest in the Inn to freeze your Dailies so they won't hurt you.
- Find Resources about HabitRPG (scroll down).

REST IN THE INN

Resting in the Inn will freeze your Dailies as-is (checked/unchecked) until the day after check-out. You will not suffer for missing them at the day's end.
Be warned: If you are participating in a boss quest, the boss will still damage you for your party mates' missed Dailies!

That might be overkill, of course (and obviously, if we change the refreshing behavior we should edit that line), but it's something to consider. At the very least, we could bold the relevant terms in the current speech bubble :)

**Comment by AnitaYCheng** [comment]
After stepping away from this thread for a while, I realized that what I said a few days ago caused an unnecessary stir in the community, and I'd like to issue a mea culpa.

I've been working on low-hanging fruit when it comes to usability for HabitRPG, and there are plenty of things to work on on a surface level. Because of this, I thought that I could hold off on doing onsite user research and checking out the Tavern and the Newbies Guild daily while working on said low-hanging fruit. That was clearly a mistake on my part, and I apologize for making assumptions. I've since made a Daily to read the Tavern and the Newbies Guild, to learn more about the community vibe. I’m still pretty new, so any pointers would be appreciated! I know I’ll still miss some things, and I really would like to learn how things work.

As for my comment about "especially subscribers," it was not my intention to make HabitRPG better for only subscribers, but I used the completely wrong phrasing and I apologize for that. One end result of HabitRPG being great (by all of our efforts) is that people love it so much they become subscribers, but getting subscribers is certainly not the only, or even the most important, end result. I just made too many assumptions with too little knowledge on my part, and am actively rectifying that now.

Definitely looking forward to continue learning about and becoming more active in the community, and to be more helpful! :smile:
